client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -65,9 +71,26 @@ export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
       return;
     }
 
-    if (mode === ModalMode.Update && nextItem) setCurrentItem(nextItem);
-    else if (mode === ModalMode.Create) setCurrentItem(null);
-    else onClose();
+    switch (true) {
+      case mode === ModalMode.Update && !!nextItem:
+        setCurrentItem(nextItem);
+        break;
+      case mode === ModalMode.Update && hasMorePages:
+        // we are at the end of the current paginated set of items
+        // fetch more pages and set the current item to null
+        // so that we can correctly set the current item when the
+        // lines query returns
+        updatePaginationQuery(page + 1);
+        setCurrentItem(null);
```

Comment:
Updating the pagination will trigger fetching the next item using `useStocktake.line.rows()`, right?

Reply:
the current item is in local state, so the updated list of items from `useStocktake.line.rows()` won't help if that's what you mean. Yes, the pagination will call the lines query and fetch items in `useStocktake.line.rows()`